### DISCUSSION TITLE
Update homebrew on macos ventura

### DISCUSSION THREAD

**Discussion by bhattpruthvi11** [Homebrew/discussions#3889] [Getting started] [answered]
Hi Team, 

I am new to mac. Earlier I installed homebrew version 2.5.0 on macos Monterey.

When apple rolled out macos ventura I updated my mac. After update, when I tried to update homebrew by 'brew upgrade' I got error as attached.
![Screenshot 2022-11-08 at 8 02 41 PM](https://user-images.githubusercontent.com/95279313/200592060-dfbd1d14-1601-413e-9928-8cc7e34b8166.JPG)

Can you please help me on:
1. Is homebrew 3.6.0 compatible with macos Ventura?

If it is compatible with new macos ventura then 
2. How to upgrade my existing brew? Because with 'brew upgrade' I am getting attached error message.

If it is not compatible then how long I have to wait for homebrew to be compatible with macos ventura..

**Comment by SMillerDev** [comment]
1. yes: https://brew.sh/2022/09/07/homebrew-3.6.0/
2. run `brew update`

**Reply by bhattpruthvi11** [reply]
Thank You! Very helpful.

**Comment by mr100percent** [comment, marked as answer]
When you upgrade the OS, it doesn't update the developer tools and compiler. You have to run the command:
`xcode-select --install`

THEN run `brew update`

**Reply by bhattpruthvi11** [reply]
Thank you.

**Reply by sepalacio** [reply]
You saved my day!

**Comment by griff-rees** [comment]
`xcode-select --install` wasn't enough for me, but `brew doctor` spotted I needed to fix my path. 

Just in case that's helpful for anyone else (below is for [`fish shell`](https://fishshell.com/), but `brew` examples should fit whichever you use):

```console
$ brew doctor
Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  fish_add_path /opt/homebrew/sbin
$ fish_add_path /opt/homebrew/sbin
Your system is ready to brew.
$ brew upgrade
```

**Comment by MarkKnochen** [comment]
mh, nothing works for me.
every brew command shows the error "Version value must be a string; got a NilClass"

i reinstalled xcode und i tried to reinstall brew, but i always get the error  "Version value must be a string; got a NilClass"

brew update shows:

/usr/local is not writable. You should change the ownership
and permissions of /usr/local back to your user account:
  sudo chown -R $(whoami) /usr/local
  
but sudo chown -R $(whoami) /usr/local shows: Operation not permitted

what can i do?

**Reply by MarkKnochen** [reply]
"brew doctor" results:

"Version value must be a string; got a NilClass"

**Reply by gromgit** [reply]
What's the output of
```
ls -ld /usr/local /usr/local/Homebrew
whoami
```
?

**Reply by MarkKnochen** [reply]
ls: /usr/local/Homebrew: No such file or directory
drwxr-xr-x  27 root  wheel  864 18 Nov 07:13 /usr/local
and my username

**Reply by gromgit** [reply]
How about the output of `id -a`?

**Comment by bhattpruthvi11** [comment]
Hi All, 

The issue is solved. When Ventura OS was rolled out for public I updated it immediately and after that I was not able to update brew as at that time Xcode was not yet released.

So, I waited for few days and when apple rolled out xcode update I updated it and then brew also got updated.

Thank you all for the help.

**Reply by MarkKnochen** [reply]
Sorry, it works not for me.

I updated the xCode in the App Store.
But all brew-commands show "Version value must be a string; got a NilClass"

brew update shows:

Error: /usr/local is not writable. You should change the ownership
and permissions of /usr/local back to your user account:
  sudo chown -R $(whoami) /usr/local
  
And: sudo chown -R $(whoami) /usr/local 
-> shows: Operation not permitted?

What i have to do?

**Reply by MarkKnochen** [reply]
this worked:

sudo chown -R $(whoami) /usr/local/*
